src/Association/Embedded.php

```diff
@@ -64,48 +66,76 @@ public function __construct($alias, $options = [])
     {
         $this->alias = $alias;
         $properties = [
-            'entityClass',
-            'property',
-            'indexClass',
+            'entityClass' => 'setEntityClass',
+            'property' => 'setProperty',
+            'indexClass' => 'setIndexClass',
         ];
         $options += [
             'entityClass' => $alias,
         ];
-        foreach ($properties as $prop) {
+        foreach ($properties as $prop => $method) {
             if (isset($options[$prop])) {
-                $this->{$prop}($options[$prop]);
+                $this->{$method}($options[$prop]);
             }
         }
     }
 
+    /**
+     * Get the property this embed is attached to.
+     *
+     * @return string The property name.
+     */
+    public function getProperty()
+    {
+        if (!$this->property) {
+            $this->property = Inflector::underscore($this->alias);
+        }
+
+        return $this->property;
+    }
+
+    /**
+     * Set the property this embed is attached to.
+     *
+     * @param string|null $name The property name to set.
+     * @return string The property name.
+     */
+    public function setProperty($name = null)
+    {
+        $this->property = $name;
+
+        return $this;
+    }
+
     /**
      * Get/set the property this embed is attached to.
      *
+     * @deprecated 2.0.1 Use setProperty()/getProperty() instead.
      * @param string|null $name The property name to set.
      * @return string The property name.
      */
     public function property($name = null)
     {
-        if ($name === null) {
-            if (!$this->property) {
-                $this->property = Inflector::underscore($this->alias);
-            }
-
-            return $this->property;
+        deprecationWarning(
+            static::class . '::property() is deprecated. ' .
+            'Use setProperty()/getProperty() instead.'
+        );
+        if ($name !== null) {
+            $this->setProperty($name);
         }
-        $this->property = $name;
+
+        return $this->getProperty();
     }
 
     /**
-     * Get/set the entity/document class used for this embed.
+     * Get the entity/document class used for this embed.
      *
-     * @param string|null $name The class name to set.
      * @return string The class name.
      */
-    public function entityClass($name = null)
+    public function getEntityClass()
```

Comment:
This method hasn't been split into getEntityClass & setEntityClass in the 3.x releases yet.